### PR TITLE
[Lake] Add publish version service for lake tablet

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [x] others

### CODE REVIEW

File: gensrc/proto/starlake.proto [hunk 1]
```diff
@@ -57,4 +60,30 @@ message TxnLogPB {
...
+message PublishVersionRequest {
+    repeated int64 tablet_ids = 1;
+    repeated int64 txn_ids = 2;
+    optional int64 base_version = 3;
+    optional int64 publish_version = 4;
+}
+
+message PublishVersionResponse {
+    optional StatusPB status = 1;
+    repeated int64 failed_tablets = 2;
```
Comment (imay) [inline gensrc/proto/starlake.proto:72]: In @decster new design, publish response should contain the max continuous version of published tablet.

Comment (sduzh) [reply gensrc/proto/starlake.proto:72]: The lake table has a different protocol.

File: gensrc/proto/starlake.proto [hunk 2]
```diff
@@ -57,4 +60,30 @@ message TxnLogPB {
     optional OpSchemaChange op_schema_change = 5;
 }
 
+message PublishVersionRequest {
+    repeated int64 tablet_ids = 1;
+    repeated int64 txn_ids = 2;
+    optional int64 base_version = 3;
```
Comment (wyb) [inline gensrc/proto/starlake.proto:66]: Why need base_version?

Comment (sduzh) [reply gensrc/proto/starlake.proto:66]: When sending a publish version task for a schema change job, the base version and the published version cae be discontinuous.

Comment (wyb) [reply gensrc/proto/starlake.proto:66]: `base_version = publish_version - txn_ids.size()`

This must be certain.

Comment (sduzh) [reply gensrc/proto/starlake.proto]: updated

File: gensrc/proto/starlake.proto [hunk 3]
```diff
@@ -57,4 +60,30 @@ message TxnLogPB {
     optional OpSchemaChange op_schema_change = 5;
 }
 
+message PublishVersionRequest {
+    repeated int64 tablet_ids = 1;
+    repeated int64 txn_ids = 2;
+    // The version to be published. base version would be version - txn_ids_size()
+    optional int64 version = 4;
```
Comment (wyb) [inline gensrc/proto/starlake.proto:67]:
```suggestion
    optional int64 version = 3;
```

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — APPROVED** [review]

File: be/src/service/service_cn/starrocks_cn.cpp [hunk 4]
```diff
@@ -34,11 +41,18 @@ void start_cn() {
     }
 
     // 2. Start brpc service.
-    std::unique_ptr<starrocks::BRpcService> brpc_service = std::make_unique<starrocks::BRpcService>(exec_env);
-    status = brpc_service->start(starrocks::config::brpc_port,
-                                 new starrocks::ComputeNodeInternalServiceImpl<starrocks::PInternalService>(exec_env),
-                                 new starrocks::ComputeNodeInternalServiceImpl<doris::PBackendService>(exec_env));
```
Comment (choury) [inline be/src/service/service_cn/starrocks_cn.cpp:40]: Why remove PBackendService from CN? this broker it.

**Pull request merged**